cmd/frontend/graphqlbackend/site_monitoring.go

```diff
@@ -0,0 +1,114 @@
+package graphqlbackend
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/inconshreveable/log15"
+	"github.com/opentracing/opentracing-go/ext"
+	prometheus "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/sourcegraph/sourcegraph/internal/trace/ot"
+)
+
+// MonitoringAlert implements the GraphQL type MonitoringAlert.
+type MonitoringAlert struct {
+	TimestampValue   DateTime
+	NameValue        string
+	ServiceNameValue string
+	OccurrencesValue int32
+}
+
+func (r *MonitoringAlert) Timestamp() DateTime { return r.TimestampValue }
+func (r *MonitoringAlert) Name() string        { return r.NameValue }
+func (r *MonitoringAlert) ServiceName() string { return r.ServiceNameValue }
+func (r *MonitoringAlert) Occurrences() int32  { return r.OccurrencesValue }
+
+func (r *siteResolver) MonitoringStatistics(ctx context.Context, args *struct {
+	Days *int32
+}) (*siteMonitoringStatisticsResolver, error) {
+	prom, err := newPrometheusQuerier()
+	if err != nil {
+		return nil, err
+	}
+	return &siteMonitoringStatisticsResolver{
+		prom:     prom,
+		timespan: time.Duration(*args.Days) * 24 * time.Hour,
+	}, nil
+}
+
+type siteMonitoringStatisticsResolver struct {
+	prom     prometheusQuerier
+	timespan time.Duration
+}
+
+const alertsResolution = 12 * time.Hour
+
+func (r *siteMonitoringStatisticsResolver) Alerts(ctx context.Context) ([]*MonitoringAlert, error) {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	span, ctx := ot.StartSpanFromContext(ctx, "site.MonitoringStatistics.alerts")
+
+	var err error
+	defer func() {
+		if err != nil {
+			ext.Error.Set(span, true)
+			span.SetTag("err", err.Error())
+		}
+		cancel()
+		span.Finish()
+	}()
+
+	results, warn, err := r.prom.QueryRange(ctx, "sum by (service_name,name)(alert_count{name!=\"\"})",
```

Comment:
nit: the following is easier to read IMO
```suggestion
	results, warn, err := r.prom.QueryRange(ctx, `sum by (service_name,name)(alert_count{name!=""})`,
```